app/models/mhv_account.rb

```diff
@@ -109,29 +112,41 @@ def create_mhv_account!
     if may_register?
       client_response = mhv_ac_client.post_register(params_for_registration)
       if client_response[:api_completion_status] == 'Successful'
+        StatsD.increment("#{STATSD_ACCOUNT_CREATION_KEY}.success")
+
         user.va_profile.mhv_ids = [client_response[:correlation_id].to_s]
         user.instance_variable_get(:@mvi).save
         self.registered_at = Time.current
         register!
       end
       # TODO: be prepared to handle or raise various exceptions
     end
+  ensure
+    StatsD.increment("#{STATSD_ACCOUNT_CREATION_KEY}.total") if may_register? || registered?
   end
 
   def upgrade_mhv_account!
-    if may_upgrade?
+    initially_premium = !may_upgrade?
+
+    unless initially_premium
       client_response = mhv_ac_client.post_upgrade(params_for_upgrade)
       if client_response[:status] == 'success'
+        StatsD.increment("#{STATSD_ACCOUNT_UPGRADE_KEY}.success")
+
         self.upgraded_at = Time.current
         upgrade!
       end
     end
   rescue Common::Exceptions::BackendServiceException => e
     if e.original_body['code'] == 155
+      initially_premium = true
       upgrade! # without updating the timestamp since account was not created at vets.gov
     else
+      StatsD.increment("#{STATSD_ACCOUNT_UPGRADE_KEY}.failure")
       raise e
     end
+  ensure
+    StatsD.increment("#{STATSD_ACCOUNT_UPGRADE_KEY}.total") unless initially_premium
```

Comment:
then the logic here becomes similar to that for create, but we additionally do not increment the total if it was already an upgraded account.

This makes it more readable and more consistent with the registration logic.